src/js/options.js

```diff
@@ -1,39 +1,20 @@
 
-function requestPermissions() {
+async function requestPermissions() {
   const checkbox = document.querySelector("#bookmarksPermissions");
   if (checkbox.checked) {
-    browser.permissions.request({permissions: ["bookmarks"]}).
-    then((response) => {
-      if (response) {
-        browser.runtime.sendMessage({ method: "resetBookmarksContext" });
-      } else {
-        checkbox.checked = false;
-      }
-    }).
-    catch((err) => {
-      return err.message;
-    });
+    const response = await browser.permissions.request({permissions: ["bookmarks"]});
+    if (!response) { checkbox.checked = false; }
```

Comment:
Rename `response` to `granted` to make it more obvious what it means, or use `if (!await ...) {` if you prefer to have smaller code.

Also add `return;`, to avoid sending a `resetBookmarksContext` event when nothing has changed.

Reply:
fixed this and squashed.